Mathlib/Analysis/Calculus/FDeriv/Basic.lean

```diff
@@ -754,31 +912,32 @@ theorem hasFDerivWithinAt_congr_set' (y : E) (h : s =ᶠ[𝓝[{y}ᶜ] x] t) :
         inter_comm] using h
     _ ↔ HasFDerivWithinAt f f' t x := hasFDerivWithinAt_diff_singleton _
 
-theorem hasFDerivWithinAt_congr_set (h : s =ᶠ[𝓝 x] t) :
+theorem hasFDerivWithinAt_congr_set [T1Space E] (h : s =ᶠ[𝓝 x] t) :
     HasFDerivWithinAt f f' s x ↔ HasFDerivWithinAt f f' t x :=
   hasFDerivWithinAt_congr_set' x <| h.filter_mono inf_le_left
```

Comment:
I suspect `T1Space` is not actually needed here, but arguably that's out of scope, since this is already much more general than what we had before!